src/com/magento/idea/magento2plugin/inspections/xml/PluginDeclarationInspection.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.inspections.xml;
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.ide.highlighter.XmlFileType;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.*;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlAttributeValue;
+import com.intellij.psi.xml.XmlDocument;
+import com.intellij.psi.xml.XmlTag;
+import com.jetbrains.php.lang.inspections.PhpInspection;
+import com.magento.idea.magento2plugin.indexes.PluginIndex;
+import com.magento.idea.magento2plugin.magento.files.ModuleDiXml;
+import com.magento.idea.magento2plugin.magento.files.ModuleXml;
+import com.magento.idea.magento2plugin.magento.packages.Package;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+
+public class PluginDeclarationInspection extends PhpInspection {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder problemsHolder, boolean b) {
+        return new XmlElementVisitor() {
+            private final String moduleXmlFileName = ModuleXml.getInstance().getFileName();
+            private static final String duplicatedPluginNameSameFileProblemDescription = "The plugin name already used in this file. For more details see Inspection Description.";
+            private static final String duplicatedPluginNameProblemDescription =
+                    "The plugin name \"%s\" for targeted \"%s\" class is already used in the module \"%s\" (%s scope). For more details see Inspection Description.";
```

Comment:
We have begun using translation bundles for phrases like https://github.com/magento/magento2-phpstorm-plugin/blob/1.0.0-develop/resources/magento2/validation.properties. Could you please create the 'inspection' bundle and move all phrases there. Thank you!

Reply:
@eduard13 really don’t like phrases within classes 😄